pfsimulator/amps/mpi1/amps_gpupacking.cu

```diff
@@ -0,0 +1,475 @@
+/**********************************************************************
+ *
+ *  Please read the LICENSE file for the GNU Lesser General Public License.
+ *
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License (as published
+ *  by the Free Software Foundation) version 2.1 dated February 1999.
+ *
+ *  This program is distributed in the hope that it will be useful, but
+ *  WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms
+ *  and conditions of the GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public
+ *  License along with this program; if not, write to the Free Software
+ *  Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307
+ *  USA
+ ***********************************************************************/
+
+extern "C"{
+
+#include <stdlib.h>
+#include <string.h>
+#include "amps.h"
+
+// #define DISABLE_GPU_PACKING
+
+#ifdef DISABLE_GPU_PACKING
+//Dummy definitions if no GPU packing
+void amps_gpu_free_bufs(){}
+
+void amps_gpu_destroy_streams(){}
+
+void amps_gpu_sync_streams(int id){
+  (void)id;
+}
+
+int amps_gpupacking(int action, amps_Invoice inv, int inv_num, char **buffer_out, int *size_out){
+  (void)action;
+  (void)inv;
+  (void)inv_num;
+  (void)buffer_out;
+  (void)size_out;
+  return 1;
+}    
+#else
+
+amps_GpuBuffer amps_gpu_recvbuf = 
+{
+  .buf = NULL, 
+  .buf_host = NULL,
+  .buf_size = NULL,
+  .num_bufs = 0
+};
+
+amps_GpuBuffer amps_gpu_sendbuf = 
+{
+  .buf = NULL, 
+  .buf_host = NULL,
+  .buf_size = NULL,
+  .num_bufs = 0
+};
+
+amps_GpuStreams amps_gpu_streams = 
+{
+  .stream = NULL,
+  .stream_id = NULL,
+  .num_streams = 0,
+  .reqs_since_sync = 0
+};
+
+/* Use page-locked host staging buffer by default */
+static int ENFORCE_HOST_STAGING = 1;
+
+/* Check for GPUDirect environment variable */
+void amps_gpu_check_env(){
```

Comment:
Functions should be documented with doxygen style comments.

Reply:
Done